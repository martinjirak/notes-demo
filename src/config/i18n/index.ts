import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import xhr from 'i18next-xhr-backend'
import { initReactI18next } from 'react-i18next'

const options: i18n.InitOptions = {
    load: 'languageOnly',
    ns: ['global'],
    fallbackLng: 'en',
    defaultNS: 'global',
    backend: {
        loadPath: 'public/locales/{{lng}}/{{ns}}.json',
    },
    detection: {
        order: ['localStorage', 'navigator'],
        caches: ['localStorage'],
        lookupLocalStorage: 'i18nextLng',
    },
    interpolation: {
        escapeValue: false,
    },
    react: {
        bindI18n: 'languageChanged',
    },
}

export default i18n.use(LanguageDetector).use(xhr).use(initReactI18next).init(options)
