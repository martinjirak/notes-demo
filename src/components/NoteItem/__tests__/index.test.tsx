import * as React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../../config/tests/i18n/index'
import { NoteItem } from '../'

describe('NoteItem should match Snapshot', () => {
    it('without hint', () => {
        const renderedValue = createMount()(
            <I18nextProvider i18n={i18n}>
                <NoteItem />
            </I18nextProvider>
        )
        expect(renderedValue.html()).toMatchSnapshot()
    })
})
