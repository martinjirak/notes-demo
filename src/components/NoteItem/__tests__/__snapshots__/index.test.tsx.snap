// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem should match Snapshot without hint 1`] = `"<div class=\\"MuiPaper-root makeStyles-root-1 MuiPaper-elevation3 MuiPaper-rounded\\"><div class=\\"makeStyles-header-2\\"><p class=\\"MuiTypography-root makeStyles-title-4 MuiTypography-body1 MuiTypography-colorPrimary\\"></p><div class=\\"makeStyles-buttons-5\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div><p class=\\"MuiTypography-root makeStyles-body-3 MuiTypography-body1\\"></p></div>"`;
